### PR TITLE
Hdrp axf v0.1.5 preview

### PR DESCRIPTION
**Pull request by slunity**
### Purpose of this PR
Many fixes for the AxF material, supported by fixes on the v0.1.5-preview of the importer

---
### Release Notes
Fix area lights to appear consistent with environments.
Fix refraction formula used.
Fix dot product clamping formulas
Fix car paint BRDF color table access. Add a mechanism to handle half-way populated tables. Required importer changes (v0.1.5-preview).
Fix (remove) gamma conversion happening on CT lobe parameters (CTF0s, CTCoeffs, CTSpreads). Required importer changes (v0.1.4-preview).
Fix BRDF enum type properties in the .shader (removes a warning on import).
Minor changes to Cook-Torrance importance sampling for pre-integrated FGD (this is just using Beckmann as the importance function instead of GGX).
Fix and document a missing factor of 4 in environments evaluation for dealing with a missing Jacobian term in the original (wrong) Cook-Torrance formulas documented in the SDK.
(the BSDF evaluation for punctual and directional lights is still using the original CT formula as we assume the error to be compensated by the presumably fitted CTCoeffs).
Document an eyeballed factor to match our LD environments (with a GGX D()) approximately to VRED.
Fix the tangent frame to make sure the AxF TBN frame follows the same convention as in the other HDRP shaders (generate the bitangent by the handedness-agnostic cross product).
Remove anisotropic rotation matrix and application at each shading sample by directly adjusting the inital TBN frame.
Fix error in anisotropic rotation handling and match the new importer code (0 to 1 is decoded to -pi/2, pi/2).
Fix anisotropic roughness parameters handling (to match, the new importer doesn't duplicate a single specularLobe texture channel in isotropic cases).
Check GGX formulas of the SDK match ours and where appropriate use those in the shared BSDF.hlsl.
Fix anisotropic GGX shadowing-masking function to use not-joint G1()G1() as specified in the SDK but considering anisotropy properly.
(Added support function GetProjectedRoughness() in BSDF for that. Also document the difference between using the projected roughness and a scalar averaged roughness in the code.)
Fix heavy flake artifacts caused by faulty flake BTF access (complete documentation of the BTF is provided in the code).
Fix artificial clearcoat dimming by a wrong division by PI in some "specular color" components.
Fix clearcoat handling (specular lobe) by a close-to-dirac lobe, similar approach to Lit, instead of an effectively diffuse component.
Fix repeated accesses to the LTC matrix for each CT lobe evaluations and pre-load those in prelightData.
Fix double BSDF / split-sum evaluation for environments, and preload per-lobe pre-integrated FGD values instead of fetching for each environment and LTC light.

Fix ConvertSurfaceDataToNormalData to ouput values according to the same heuristic as StackLit.
Add SSR lighting.

Add SSR opt-out option.
Add decal opt-out option and fix stencil tagging.

TODO:

-expose control for light type intensity factors (?)
-anisotropic IBL hack
-non-refracted directions for both LD fetches for environments and LTC irradiance (ie for split sum lights)
-proper dual normal handling for punctual and directional lights evaluation (cf StackLit)

---
### Testing status
**Katana Tests**: First off we need to make sure the Katana SRP tests are green?

**Manual Tests**: What did you do?

**Automated Tests**: What did you setup?

Any test projects to go with this to help reviewers?

---
### Overall Product Risks
**Technical Risk**: None, Low, Medium, High?
Low, affects just AxF material.

**Halo Effect**: None, Low, Medium, High?

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/Material/AxF/AxF.hlsl [hunk 1]
```diff
@@ -224,11 +315,11 @@ float CT_F(float H_V, float F0)
 float3  MultiLobesCookTorrance(float NdotL, float NdotV, float NdotH, float VdotH)
 {
     // Ensure numerical stability
-    if (NdotV < 0.00174532836589830883577820272085 && NdotL < 0.00174532836589830883577820272085) //sin(0.1°)
+    if (NdotV < 0.00174532836589830883577820272085 || NdotL < 0.00174532836589830883577820272085) //sin(0.1°)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Material/AxF/AxF.hlsl]: should use a define for this constant instead :)

**Comment by sebastienlagarde** [comment]
Please update changelog.md (with few line (2 or 3) + launch Katana test

**Comment by sebastienlagarde** [comment]
I launched them for you, next time launch them :)

https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=hdrp-axf-v0.1.5-preview&automation-tools_branch=add-platform-filter&unity_branch=trunk#

If green, I will merge the PR

File: com.unity.render-pipelines.high-definition/Runtime/Material/AxF/AxF.hlsl [hunk 2]
```diff
@@ -1579,18 +1940,58 @@ IndirectLighting    EvaluateBSDF_ScreenspaceRefraction( LightLoopContext lightLo
...
+        // Also it appear like more sharp. To avoid these artifact and at the same time get better match to reference we lerp to original unmodified reflection.
+        // Formula is empirical.
+        ret = GetSpecularDominantDir(N, iblR, iblPerceptualRoughness, clampedNdotV);
+        float iblRoughness = PerceptualRoughnessToRoughness(iblPerceptualRoughness);
+        ret = lerp(ret, iblR, saturate(smoothstep(0, 1, iblRoughness * iblRoughness)));
+    }
+    return ret;
+}
+
+IndirectLighting EvaluateBSDF_Env_e(  LightLoopContext lightLoopContext,
```
[the remark this answers is not in the document]
Comment (slunity) [reply com.unity.render-pipelines.high-definition/Runtime/Material/AxF/AxF.hlsl]: Yes. Was just used to test when working on env.

**Pull request merged**